src/session/components/avatar.rs

```diff
@@ -111,74 +124,127 @@ impl Avatar {
-            gtk::ClosureExpression::new::<Option<gtk::gdk::Paintable>, _, _>(
-                &[
-                    icon_name_expression.clone().upcast(),
-                    custom_image_expression.upcast(),
-                ],
-                closure!(|_: Self,
-                          icon_name: Option<String>,
-                          cusom_image: Option<gtk::gdk::Paintable>| {
-                    match icon_name {
-                        Some(_) => None,
-                        None => cusom_image,
-                    }
-                }),
-            );
-
-        let show_initials_expression = icon_name_expression.chain_closure::<bool>(closure!(
-            |_: Self, icon_name: Option<String>| icon_name.is_none()
-        ));
-
-        let text_expression = item_expression.chain_property::<AvatarItem>("display-name");
-        let text_expression = gtk::ClosureExpression::new::<Option<String>, _, _>(
-            &[
-                icon_name_expression.clone().upcast(),
-                text_expression.upcast(),
-            ],
-            closure!(
-                |_: Self, icon_name: Option<String>, display_name: Option<String>| {
-                    icon_name
-                        // If we use an icon for the avatar, we always want its color to be blue. As
-                        // AdwAvatar doesn't allow us to set the color in an explicit manner, we are
-                        // forced to use this workaround.
-                        .map(|_| Some(String::from("-")))
-                        .unwrap_or(display_name)
+        // Chat title expression
+        let title_expression = item_expression.chain_property::<Chat>("title");
+        gtk::ClosureExpression::new::<String, _, _>(
+            &[item_expression.clone(), title_expression],
+            closure!(|_: Self, chat: Chat, title: String| {
+                if chat.is_own_chat() {
+                    // Workaround for having a blue AdwAvatar for Saved Messages chat
+                    "-".to_string()
+                } else {
+                    title
                 }
-            ),
-        );
+            }),
+        )
+        .bind(&*imp.avatar, "text", Some(self));
 
-        let imp = self.imp();
+        // User title expression
+        User::full_name_expression(&item_expression).bind(&*imp.avatar, "text", Some(self));
 
+        // Icon expression
+        let icon_name_expression =
+            item_expression.chain_closure::<Option<String>>(closure!(|_: Self,
+                                                                      item: Option<
+                glib::Object,
+            >| {
+                if let Some(Ok(chat)) = item.map(|i| i.downcast::<Chat>()) {
+                    if chat.is_own_chat() {
+                        // Show bookmark icon for Saved Messages chat
+                        return Some("user-bookmarks-symbolic".to_string());
+                    }
+                }
+
+                None
```

Comment:
This does not seem to work. I fail to understand why, but the bookmark avatar does not appear on my side. I noticed that a more functional style works though. Weird… 
```suggestion
                item.as_ref()
                    .and_then(|i| i.downcast_ref::<Chat>())
                    .map(Chat::is_own_chat)
                    // Show bookmark icon for Saved Messages chat
                    .map(|_| "user-bookmarks-symbolic")
```